app/middleware/auth.py

```diff
@@ -69,20 +86,24 @@ async def auth_dependency(request: Request, user_status: UserStatus):
             raise HTTPException(status_code=403, detail="User is not logged in")
 
 
-@cache(
-    key_builder=generate_cache_key,
-    namespace="user-data",
-    expire=60 * 60,
-)
 async def get_user_data(session_cookie: str, auth_base_url: str) -> dict:
-    print(session_cookie, auth_base_url)
+    if not session_cookie:
+        return await _fetch_user_data(session_cookie, auth_base_url)
+    return await _get_user_data_cached(session_cookie, auth_base_url)
```

Comment:
`session_cookie` is necessarily a non-empty string at this point. I believe you should only have `return await _get_user_data_cached(session_cookie, auth_base_url)` at this point (or even better, directly call this in `auth_dependency`